docs/hugo/content/design/ADR-2022-11-Change-Detection.md

```diff
@@ -4,79 +4,115 @@ title: '2022-11: Change Detection'
 
 ## Context
 
-In ASO v2 up to at least the `beta.4` release, we reconcile each resource by doing a PUT, relying on the Azure Resource Manager (ARM) to do the goal state comparison and only update the resource if it has changed. 
+In ASO v2 up to at least the v2.15 release, we reconcile each resource by performing a PUT operation, relying on the Azure Resource Manager (ARM) to do goal state comparison and to only update the resource if it has changed.
 
-While this works, customers are already running up against severe ARM throttling with moderate numbers of resources. Typically, ARM throttles PUT requests handled by a given ARM endpoint to just 1200 per hour per subscription. With our current use of a single active HTTP connection and a reconcile period of 15m, ASO users are hitting this limit with just 300 active resources.
+While this approach works, some customers are running into ARM throttling limits with moderate numbers of resources. This occurs despite ARM throttling limits being lifted substantially since ASO went GA.
```

Comment:
Customers also sometimes run into issues where resources themselves are not idempotent (AKS AgentPool is an example of this although I am not sure we've had many complaints).